test/functional/feature_datacarrier.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test datacarrier functionality"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.script import (
+    CScript,
+    OP_RETURN,
+)
+from test_framework.wallet import (
+    MiniWallet
+)
+from test_framework.messages import (
+    CTxOut,
+)
+from test_framework.util import (
+    assert_raises_rpc_error,
+)
+
+class DataCarrierTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def test_transaction(self, tx, success):
+        tx_hex = tx.serialize().hex()
+
+        if success:
+            self.wallet.sendrawtransaction(from_node=self.nodes[0], tx_hex=tx_hex)
+            assert(tx.rehash() in self.nodes[0].getrawmempool(True))
+        else:
+            assert_raises_rpc_error(-26, "scriptpubkey", self.wallet.sendrawtransaction, from_node=self.nodes[0], tx_hex=tx_hex)
+
+    def create_null_data_transaction(self, script):
+        tx = self.wallet.create_self_transfer(fee_rate=0)["tx"]
+        tx.vout.append(CTxOut(nValue=0, scriptPubKey=script))
+        tx.vout[0].nValue -= tx.get_vsize()  # simply pay 1sat/vbyte fee
+        return tx
+
+    def run_test(self):
+        self.node = self.nodes[0]
+        self.wallet = MiniWallet(self.nodes[0])
+        self.wallet.rescan_utxos()
+
+        self.log.info("Testing null data transaction with default -datacarrier and -datacarriersize values.")
+        script = CScript([OP_RETURN, b"Hello world!"])
+        tx = self.create_null_data_transaction(script)
+
+        self.test_transaction(tx, success=True)
+
+        self.log.info("Testing a null data transaction larger than allowed by the default -datacarriersize value.")
+        data = b"Hello world!" * 10
```

Comment:
`data = b"Hello world!" * 7` would have worked just as well here.

I'd recommend making these values just above the limits to detect off-by-one errors. For example, instead of composing it of "hello world" strings, use a `MAX_OP_RETURN_RELAY`-length string.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25792/commits/10bfe1204fed71ff10d6bbac90912804894d454b. Thanks for review.